pandas/core/frame.py

```diff
@@ -2121,6 +2122,13 @@ def eval(self, expr, **kwargs):
         ----------
         expr : string
             The expression string to evaluate.
+        inplace : bool
+            If the expression contains an assignment, whether to return a new
+            DataFrame or mutate the existing.
+
+            WARNING: inplace=None currently falls back to to True, but
```

Comment:
I don't really like it (something about filtering feels like it shouldn't be in-place) but it would definitely be more consistent - `.where` and the like have that option